homeassistant/components/victron_gx_mqtt/__init__.py

```diff
@@ -0,0 +1,144 @@
+    from homeassistant.core import HomeAssistant, ServiceCall
+    from homeassistant.helpers.typing import ConfigType
+
+from homeassistant.helpers import config_validation as cv
+
+from .const import ATTR_DEVICE_ID, ATTR_METRIC_ID, ATTR_VALUE, DOMAIN, SERVICE_PUBLISH
+from .hub import Hub
+
+_LOGGER = logging.getLogger(__name__)
+_VICTRON_MQTT_LOGGER = logging.getLogger("victron_mqtt")
+
+# Config schema - this integration is config entry only
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+PLATFORMS: list[Platform] = [
+    Platform.SENSOR,
+]
+
+__all__ = ["DOMAIN"]
+
+
+async def async_setup_services(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Set up services for the Victron MQTT integration."""
+
+    # Only register services once
+    if hass.services.has_service(DOMAIN, SERVICE_PUBLISH):
+        return
+
+    async def handle_publish(call: ServiceCall) -> None:
+        """Handle the set_value service call."""
+        metric_id = call.data.get(ATTR_METRIC_ID)
+        device_id = call.data.get(ATTR_DEVICE_ID)
+        value = call.data.get(ATTR_VALUE)
+
+        if not metric_id:
+            raise HomeAssistantError("metric_id is required")
+        if not device_id:
+            raise HomeAssistantError("device_id is required")
+
+        # Find the hub instance
+        hub: Hub = entry.runtime_data
+        if hub is None:
+            raise HomeAssistantError("No Victron MQTT hub found")
+
+        hub.publish(metric_id, device_id, value)
+
+    # Register the service
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_PUBLISH,
+        handle_publish,
+    )
+
+    _LOGGER.info("Victron MQTT services registered")
+
+
+async def _update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    _LOGGER.info("Options for victron_mqtt have been updated - applying changes")
+    # Reload the integration to apply changes
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
+async def get_package_version(package_name: str) -> str:
+    return await asyncio.get_event_loop().run_in_executor(
+        None, importlib.metadata.version, package_name
+    )
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up the integration."""
+    version = getattr(hass.data["integrations"][DOMAIN], "version", 0)
+    victron_mqtt_version = await get_package_version("victron_mqtt")
+    _LOGGER.info(
+        "Setting up victron_mqtt integration. Version: %s. victron_mqtt package version: %s",
+        version,
+        victron_mqtt_version,
+    )
+
+    return True
+
+
+def _sync_library_logging():
+    """Sync the log level of the library to match integration logging."""
+    lib_level = _LOGGER.getEffectiveLevel()
+    _VICTRON_MQTT_LOGGER.setLevel(lib_level)
+    _VICTRON_MQTT_LOGGER.propagate = True  # Let it go through HA logging
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
Extend the config entry type to add the thing you'll add in runtime_data in the typslot, check the `runtime-data` rule for an example